src/main/kotlin/app/revanced/patches/youtube/layout/spoofappversion/SpoofAppVersionPatch.kt

```diff
@@ -42,42 +42,37 @@ object SpoofAppVersionPatch : BytecodePatch(
-                StringResource("revanced_spoof_app_version_user_dialog_message",
-                "App version will be spoofed to an older version of YouTube."
-                        + "\\n\\nThis will change the appearance and features of the app, but unknown side effects may occur."
-                        + "\\n\\nIf later turned off, it is recommended to clear the app data to prevent UI bugs.")
+                "revanced_spoof_app_version_title",
+                "revanced_spoof_app_version_summary_on",
+                "revanced_spoof_app_version_summary_off"
             ),
             ListPreference(
                 "revanced_spoof_app_version_target",
-                StringResource(
-                    "revanced_spoof_app_version_target_title",
-                    "Spoof app version target"
-                ),
+                "revanced_spoof_app_version_target_title",
                 ArrayResource(
                     "revanced_spoof_app_version_target_entries",
                     listOf(
-                        StringResource("revanced_spoof_app_version_target_entry_1", "18.33.40 - Restore RYD Shorts incognito mode"),
-                        StringResource("revanced_spoof_app_version_target_entry_2", "18.20.39 - Restore wide video speed & quality menu"),
-                        StringResource("revanced_spoof_app_version_target_entry_3", "17.08.35 - Restore old UI layout"),
-                        StringResource("revanced_spoof_app_version_target_entry_4", "16.08.35 - Restore explore tab"),
-                        StringResource("revanced_spoof_app_version_target_entry_5", "16.01.35 - Restore old Shorts player"),
+                        "revanced_spoof_app_version_target_entry_1",
+                        "revanced_spoof_app_version_target_entry_2",
+                        "revanced_spoof_app_version_target_entry_3",
+                        "revanced_spoof_app_version_target_entry_4",
+                        "revanced_spoof_app_version_target_entry_5",
                     )
                 ),
                 ArrayResource(
                     "revanced_spoof_app_version_target_entry_values",
                     listOf(
-                        StringResource("revanced_spoof_app_version_target_entry_value_1", "18.33.40"),
-                        StringResource("revanced_spoof_app_version_target_entry_value_2", "18.20.39"),
-                        StringResource("revanced_spoof_app_version_target_entry_value_3", "17.08.35"),
-                        StringResource("revanced_spoof_app_version_target_entry_value_4", "16.08.35"),
-                        StringResource("revanced_spoof_app_version_target_entry_value_5", "16.01.35"),
-                    )
+                        "18.33.40",
+                        "18.20.39",
+                        "17.08.35",
+                        "16.08.35",
+                        "16.01.35",
+                    ),
+                    literalValues = true
```

Comment:
> is typos will be treated at literals

Another option would be to simply input the full string value. This means, if you want to use a literal, you'd enter a literal value, but if you want to use a reference, you'd input `@string/value`. This prevents having to create boilerplate classes, manually specifying if the values are literals and allows for each entry to be a reference or literal.

Reply:
> That would mean boilerplate @string/ prefix is needed. 

Yes. A certain amount of boilerplate will be necessary. May it be in the form of abstraction or treating the attribute value literally. It would certainly get rid of the parameter to specify if the inputs are literals as well as allow each value to be literal or reference though. The added boilerplate is minimal though and straight forward to use whereas I had to first understand why a parameter to specify the values as literals exists.

>  And it's not consistent with where keys are used everywhere else without declaring it's for a string resource.

Keys are different from values, the parameters ask for a key specifically, but here we ask for values, or am I understanding you incorrectly?

Reply:
Yeah, that way my initial solution above too but it would require the user to wrap every string in one, which is cumbersome, I'd rather leave it as it is right now, otherwise, if you want, we can use the abstraction